File: src/com/atslangplugin/ATSFileTypeStatic.java
package com.atslangplugin;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * Created by brandon on 12/16/14.
 */
public class ATSFileTypeStatic extends LanguageFileType {
    public static final ATSFileTypeStatic INSTANCE = new
            ATSFileTypeStatic();

    private ATSFileTypeStatic() {
        super(ATSLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "sats file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Static Applied Type System (ATS) language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "sats";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return ATSIcons.FILE;
    }
}

File: src/com/atslangplugin/ATSFileTypeStatic.kt
package com.atslangplugin

import com.intellij.openapi.fileTypes.LanguageFileType

import javax.swing.*

/**
 * Created by brandon on 12/16/14.
 */
public class ATSFileTypeStatic private() : LanguageFileType(ATSLanguage.INSTANCE) {

    override fun getName(): String {
        return "sats file"
    }

    override fun getDescription(): String {
        return "Static Applied Type System (ATS) language file"
    }

    override fun getDefaultExtension(): String {
        return "sats"
    }

    override fun getIcon(): Icon? {
        return ATSIcons.FILE
    }

    companion object {
        public val INSTANCE: ATSFileTypeStatic = ATSFileTypeStatic()
    }
}
